### PR TITLE
ENT-11385 Fix compiler warnings

### PR DESCRIPTION
**Pull request by balwantkothari**
Fix compiler warnings for JDK17

### CODE REVIEW

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt [hunk 1]
```diff
@@ -3,17 +3,16 @@ package com.r3.corda.lib.reissuance.contracts
 import com.r3.corda.lib.reissuance.states.ReissuanceLock
 import com.r3.corda.lib.reissuance.states.ReissuanceRequest
 import net.corda.core.contracts.*
+import net.corda.core.crypto.*
```
Comment (adelel1) [inline contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:6]: Can you explicitly specify the imports.

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt [hunk 2]
```diff
@@ -196,7 +195,7 @@ class ReissuanceLockContract<T>: Contract where T: ContractState {
 
         requireThat {
             "Exactly one input of type ReissuanceLock is expected" using (reissuanceLockInputs.size == 1)
-            val reissuanceLockInput = reissuanceLockInputs[0].state.data as ReissuanceLock<T>
+            val reissuanceLockInput = reissuanceLockInputs[0].state.data as ReissuanceLock<*>
```
Comment (adelel1) [inline contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:198]: Revert and add a @Suppress("UNCHECKED_CAST")

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt [hunk 3]
```diff
@@ -247,14 +246,14 @@ class ReissuanceLockContract<T>: Contract where T: ContractState {
         val availableComponentNonces: Map<Int, List<SecureHash>> by lazy {
             wireTransaction.componentGroups.map { Pair(it.groupIndex, it.components.mapIndexed {
                 internalIndex, internalIt ->
-                componentHash(internalIt, wireTransaction.privacySalt, it.groupIndex, internalIndex) })
+                digestService.componentHash(internalIt, wireTransaction.privacySalt, it.groupIndex, internalIndex) })
```
Comment (adelel1) [inline contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:249]: Use DigestService.default
Save having to create a new object.

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt [hunk 4]
```diff
@@ -247,14 +246,14 @@ class ReissuanceLockContract<T>: Contract where T: ContractState {
...
             }.toMap()
         }
 
         val availableComponentHashes = wireTransaction.componentGroups.map {
             Pair(it.groupIndex, it.components.mapIndexed {
                 internalIndex, internalIt ->
-                componentHash(availableComponentNonces[it.groupIndex]!![internalIndex], internalIt) })
+                digestService.componentHash(availableComponentNonces[it.groupIndex]!![internalIndex], internalIt) })
```
Comment (adelel1) [inline contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:256]: Use DigestService.default.

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt (hunk 1 above)
Comment (balwantkothari) [reply contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:6]: Done.

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt (hunk 2 above)
Comment (balwantkothari) [reply contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:198]: Done.

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt (hunk 4 above)
Comment (balwantkothari) [reply contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:256]: Done.

File: contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt (hunk 3 above)
Comment (balwantkothari) [reply contracts/src/main/kotlin/com/r3/corda/lib/reissuance/contracts/ReissuanceLockContract.kt:249]: Done.

**Review by adelel1 — APPROVED** [review]

**Pull request merged**